### PR TITLE
OSDOCS-5260:Etcd docs for MicroShift

### PR DESCRIPTION
**Pull request by opayne1**
[OSDOCS-5260](https://issues.redhat.com//browse/OSDOCS-5260):Etcd docs for MicroShift

Version(s):
4.13+

Issue:
https://issues.redhat.com/browse/OSDOCS-5260

Link to docs preview:
https://56833--docspreview.netlify.app/microshift/latest/microshift_support/microshift-etcd.html

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: microshift_support/microshift-etcd.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+[id="etcd-microshift"]
+= MicroShift etcd systemd log stream
+//another name idea could be "Etcd as a transient systemd unit that is managed by MicroShift"
+include::_attributes/attributes-microshift.adoc[]
+include::_attributes/common-attributes.adoc[]
+:context: etcd-microshift
+
+toc::[]
+
+MicroShift etcd logs will operate in their own system log stream by running the etcd container image under Podman and running the etcd binary installed on a MicroShift RPM.
```
Comment (dhellmann) [inline microshift_support/microshift-etcd.adoc:11]: etcd does not run in a container, so podman is not used. It runs as a transient systemd unit, and that's what ensures the logs are separated. However, I don't think we want user documentation to go into that level of detail about implementation. I think it is sufficient to say something like "MicroShift etcd runs as a separate process and the logs will appear in their own system log stream called name-goes-here."

File: microshift_support/microshift-etcd.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+:FeatureName: MicroShift
+include::snippets/technology-preview.adoc[leveloffset=+1]
+//
+
+As a part of MicroShift installation, etcd will automatically start by using `systemd-run etcd`. You will be able to view etcd running under systemd `systemctl status microshift-etcd` and collect logs from it.
```
Comment (dhellmann) [inline microshift_support/microshift-etcd.adoc:17]: The MicroShift process manages the etcd process. How it does that is an implementation detail, so let's leave that out of these docs.

```suggestion
MicroShift manages the etcd process automatically. You will be able to view etcd running under systemd `systemctl status microshift-etcd` and collect logs from it.
```

**Review by dhellmann — COMMENTED** [review]
Tagging @dusk125 and @pmtk as SMEs for this one.

File: modules/microshift-observe-debug-etcd-server.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+:_content-type: PROCEDURE
+[id="microshift-observe-debug-etcd-server_{context}"]
+= Observe and debug the {product-title} etcd server
+
+You can gather `journalctl` logs to observe and debug the etcd server logs.
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (oc).
```
Comment (dhellmann) [inline modules/microshift-observe-debug-etcd-server.adoc:13]: `oc` is not used in this procedure, so I don't think it's needed.

File: microshift_support/microshift-etcd.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+toc::[]
+
+:FeatureName: MicroShift
+include::snippets/microshift-tech-preview-snip.adoc[leveloffset=+1]
+
+[role="_abstract"]
+{product-title} etcd is delivered as part of the {product-title} RPM and runs as a separate process. The logs will appear in their own system log stream `journalctl -u microshift-etcd.scope`.
+
+{product-title} manages the etcd process automatically. You will be able to view etcd running under systemd `systemctl status microshift-etcd` and collect logs from it.
```
Comment (pmtk) [inline microshift_support/microshift-etcd.adoc:15]:
```suggestion
{product-title} manages the etcd process automatically. You will be able to view etcd running under systemd `systemctl status microshift-etcd.scope` and collect logs from it.
```

File: modules/microshift-observe-debug-etcd-server.adoc [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+* You have run the command `systemctl start microshift` as part of the {product-title} start-up.
+
+.Procedure
+
+* To get the logs for etcd, run the following command:
++
+[source,terminal]
+----
+$ journalctl -u microshift-etcd
```
Comment (pmtk) [inline modules/microshift-observe-debug-etcd-server.adoc:23]:
```suggestion
$ journalctl -u microshift-etcd.scope
```
Also, it needs to run as privileged user ( `$` suggests regular user, is that the case?)

File: modules/microshift-observe-debug-etcd-server.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+[id="microshift-observe-debug-etcd-server_{context}"]
+= Observe and debug the {product-title} etcd server
+
+You can gather `journalctl` logs to observe and debug the etcd server logs.
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (oc).
+
+* You have run the command `systemctl start microshift` as part of the {product-title} start-up.
```
Comment (ggiguash) [inline modules/microshift-observe-debug-etcd-server.adoc:15]: I would say that we need to have MicroShift service running as a prerequisite without mentioning the command. There is more than one way to start the service, so mentioning something concrete here is a bit misleading.

File: modules/microshift-observe-debug-etcd-server.adoc (hunk 5 above)
Comment (ggiguash) [reply modules/microshift-observe-debug-etcd-server.adoc:23]: Yes, the command must be `$ sudo journalctl -u microshift-etcd.scope`

File: modules/microshift-observe-debug-etcd-server.adoc [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+* To get the logs for etcd, run the following command:
++
+[source,terminal]
+----
+$ journalctl -u microshift-etcd
+----
++
+[NOTE]
+====
+etcd logs are not with {product-title} logs in `journalctl -u microshift`. etcd logs are found in `journalctl -u microshift-etcd.scope`.
```
Comment (ggiguash) [inline modules/microshift-observe-debug-etcd-server.adoc:28]: I do not think this note should repeat both commands. 
```suggestion
{product-title} logs can be accessed separately from etcd logs using the `sudo journalctl -u microshift` command
```

**Comment by ggiguash** [comment]
@opayne1, FYI I'm fine with the current contents.

File: microshift_support/microshift-etcd.adoc [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+include::_attributes/attributes-microshift.adoc[]
+:context: microshift-etcd
+
+toc::[]
+
+:FeatureName: MicroShift
+include::snippets/microshift-tech-preview-snip.adoc[leveloffset=+1]
+
+[role="_abstract"]
+{product-title} etcd is delivered as part of the {product-title} RPM. The etcd service is run as a separate process and its lifecycle is managed automatically by {product-title}.
```
Comment (kelbrown20) [inline microshift_support/microshift-etcd.adoc:13]: Up to you! 
```suggestion
{product-title} etcd is a part of the {product-title} RPM. The etcd service is run as a separate process and the lifecycle is managed automatically by {product-title}.
```

File: microshift_support/microshift-etcd.adoc [hunk 9]
```diff
@@ -0,0 +1,15 @@
+:_content-type: ASSEMBLY
+[id="microshift-etcd"]
+= MicroShift etcd
+include::_attributes/attributes-microshift.adoc[]
+:context: microshift-etcd
+
+toc::[]
+
+:FeatureName: MicroShift
```
Comment (kelbrown20) [inline microshift_support/microshift-etcd.adoc:9]: I'm noticing that throughout the docs we have the tech preview snippet in different places. Sometimes its at the beginning of the assembly and sometimes its after the intro. I'm wondering what you think looks better?

**Review by kelbrown20 — COMMENTED** [review]
Hey Olivia! Everything looks great! I just made a small comment with a question, but other wise this LGTM

File: microshift_support/microshift-etcd.adoc (hunk 9 above)
Comment (opayne1) [reply microshift_support/microshift-etcd.adoc:9]: Good question @kelbrown20 ! Lets mention it in the docs call in a bit.

**Pull request merged**